tldr/mcp_server.py

```
@@ -37,11 +44,29 @@ def _get_lock_path(project: str) -> Path:
     return Path(tmp_dir) / f"tldr-{hash_val}.lock"
 
 
+def _get_connection_info(project: str) -> tuple[str, int | None]:
+    """Return (address, port) - port is None for Unix sockets.
+
+    On Windows, uses TCP on localhost with a deterministic port.
+    On Unix, uses Unix domain sockets.
+    """
+    if sys.platform == "win32":
+        hash_val = hashlib.md5(str(Path(project).resolve()).encode()).hexdigest()[:8]
```

Ruff (0.14.10)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)